hathor/cli/db_export.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import struct
+from argparse import ArgumentParser, FileType, Namespace
+from typing import Iterator, Optional
+
+from hathor.cli.run_node import RunNode
+
+MAGIC_HEADER = b'HathDB'
+
+
+class DbExport(RunNode):
+    def start_manager(self, args: Namespace) -> None:
+        pass
+
+    def register_signal_handlers(self, args: Namespace) -> None:
+        pass
+
+    def register_resources(self, args: Namespace) -> None:
+        pass
+
+    def create_parser(self) -> ArgumentParser:
+        from hathor.conf import HathorSettings
+        settings = HathorSettings()
+
+        def max_height(arg: str) -> Optional[int]:
+            if arg.lower() == 'checkpoint':
+                if not settings.CHECKPOINTS:
+                    raise ValueError('There are no checkpoints to use')
+                return settings.CHECKPOINTS[-1].height
+            elif arg:
+                return int(arg)
+            else:
+                return None
+
+        parser = super().create_parser()
```

Comment:
Should we allow skipping voided blocks and transactions?

Reply:
I see how that can be useful, I'll add that, although I won't skip soft-voided txs.

Reply:
Makes sense.

Reply:
I've added it, I made an export of our latest snapshot while excluding voided txs and I'm waiting for the import to finish to check if the soft-voided txs were correctly handled.

Reply:
Yes, it worked well.